### PR TITLE
New version of video expandable ad

### PR DESCRIPTION
**Pull request by tijanamiletic**
New version of the video expandable DFP ad template with options for right aligning the video and shrinking to 768px width while allowing for a background image which will right align to available space on smaller screens.

Wide unexpanded:

screen shot 2015-09-15 at 16 04 15

Wide expanded & playing video:

screen shot 2015-09-15 at 16 04 38

Desktop unexpanded:

screen shot 2015-09-15 at 16 05 12

Desktop expanded & playing video:

screen shot 2015-09-15 at 16 05 01

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    'common/utils/template',
+    'common/views/svgs',
+    'text!common/views/commercial/creatives/expandable-video-v2.html'
+], function (
+    bean,
+    bonzo,
+    fastdom,
+    _,
+    $,
+	detect,
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: Little alignment issue on the line 'detect'

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            this.closedHeight = 250;
+            this.openedHeight = 500;
+        } else {
+            this.closedHeight = 150;
+            this.openedHeight = 300;
+        }
+
+    };
+
+    ExpandableVideo.prototype.create = function () {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: This is simply a matter of style, and probably doesn't matter for creative template work, but I'm not sure we should have 'create' methods - 'new' should create a readied object. If a constructor gets too long, it's preferable to decompose it into private functions.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        var videoHeight = this.openedHeight,
+            showmoreArrow = {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: I know this isn't your pattern, but I'm not sure if I'm totally convinced about this house style of single var declarations - large assignments like these can get a bit unwieldy, I think. (Just my 2p)

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+                    if (this.params.trackingPixel) {
+                        this.$adSlot.before('<img src="' + this.params.trackingPixel + this.params.cacheBuster + '" class="creative__tracking-pixel" height="1px" width="1px"/>');
+                    }
+                    $ExpandableVideo.appendTo(this.$adSlot);
+                    resolve();
+                }.bind(this));
+            }.bind(this));
+
+        bean.on(this.$adSlot[0], 'click', '.ad-exp__open', function () {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: I'm not sure if the relationship between the event and the script is totally transparent. I'm a big fan of using data attributes and bindings - or at least 'js-' prefixed classes - to mark elements that will handle and invoke events. This makes it harder to break the scripted behaviour when trying to modify the HTML. Just my opinion.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            showmorePlus = {
+                showPlus: (this.params.showMoreType === 'plus-only' || this.params.showMoreType === 'plus-and-arrow') ?
+                    '<button class="ad-exp__close-button ad-exp__open">' + svgs('closeCentralIcon') + '</button>' : ''
+            },
+            videoSource = {
+                videoEmbed: (this.params.YoutubeVideoURL !== '') ?
+                    '<iframe id="YTPlayer" width="100%" height="' + videoHeight + '" src="' + this.params.YoutubeVideoURL + '?showinfo=0&amp;rel=0&amp;controls=0&amp;fs=0&amp;title=0&amp;byline=0&amp;portrait=0" frameborder="0" class="expandable-video"></iframe>' : ''
+            },
+            $ExpandableVideo = $.create(template(ExpandableVideoTpl, { data: _.merge(this.params, showmoreArrow, showmorePlus, videoSource) })),
+            domPromise = new Promise(function (resolve) {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: Again, not your style - ours - but should we be using fastdom by default? Using fastdom for things we haven't profiled means adding a lot of boilerplate to deal with asynchronicity. It also makes debugging trickier. I guess you could argue that all these DOM operations add up, so using fastdom as our first choice heads those problems off right from the start, but still... it seems to make DOM code look a lot more complicated than its actual behaviour really warrants. 

Then again, it's late... perhaps I'll see things differently tomorrow morning!

**Comment by jbreckmckye** [comment]
It's late so I should probably continue looking in the morning. The ad itself looks great, by the way.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+	detect,
+    mediator,
+    storage,
+    template,
+    svgs,
+    ExpandableVideoTpl
+) {
+
+    /**
+     * https://www.google.com/dfp/59666047#delivery/CreateCreativeTemplate/creativeTemplateId=10028247
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: What exactly is this comment? Is this the creative as it's uploaded to DFP itself?

File: static/src/javascripts/projects/common/views/commercial/creatives/expandable-video-v2.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
+<div class="creative--expandable-video <%=data.videoOptions%>">
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/views/commercial/creatives/expandable-video-v2.html]: This seems very dense... I don't know if we could perhaps mitigate this by splitting it into multiple partials or maybe at least just adding some whitespace and some comments annotating the different parts of the template.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js (hunk 3 above)
Comment (gklopper) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: I never liked those either.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js (hunk 1 above)
Comment (tijanamiletic) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: Spacing is correct in file. For some reason it doesn't align on Github.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js (hunk 2 above)
Comment (tijanamiletic) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: I'm reusing an existing template and using the style which is used elsewhere.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js (hunk 3 above)
Comment (tijanamiletic) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: I have been asked to use this standard in the past, which I think it's used throughout the site, so it's not my call to change this.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js (hunk 6 above)
Comment (tijanamiletic) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: It's now removed.

File: static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js (hunk 4 above)
Comment (tijanamiletic) [inline static/src/javascripts/projects/common/modules/commercial/creatives/expandable-video-v2.js]: Again I'm only reusing an existing template style.

File: static/src/javascripts/projects/common/views/commercial/creatives/expandable-video-v2.html (hunk 7 above)
Comment (tijanamiletic) [inline static/src/javascripts/projects/common/views/commercial/creatives/expandable-video-v2.html]: I have added some space.

**Comment by jbreckmckye** [comment]
Seems alright to me; I do think there might be some practices we could consider for future templates, but we can think about them later.

:+1:

**Pull request merged**